### PR TITLE
Updated dartdoc to 0.36.0

### PR DESCRIPTION
**Pull request by jcollins-g**
## Description

This PR updates dartdoc to 0.36.0.  See the release notes below, but TL;DR for Flutter is several bugs were fixed that will correct some broken links and make it easier to refactor code.

## Related Issues

Release notes:  https://github.com/dart-lang/dartdoc/releases/tag/v0.36.0

## Tests

Ran bot script and spot checked to be sure null safety tag still was in place and no new warnings were generated.

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -139,7 +139,6 @@ Future<void> main(List<String> arguments) async {
   final List<String> dartdocArgs = <String>[
     ...dartdocBaseArgs,
     '--allow-tools',
-    '--enable-experiment=non-nullable',
```
Comment (gspencergoog) [inline dev/tools/dartdoc.dart:142]: Why did you need to disable this?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (jcollins-g) [reply dev/tools/dartdoc.dart:142]: See the release notes -- this flag is on by default now.
